simulator.py

```diff
@@ -618,71 +674,92 @@ def at(self, location, city, duration):
             t_overlap = min(self.leaving_time, getattr(h, "leaving_time", 60)) - max(self.start_time, getattr(h, "start_time", 60))
             t_near = self.rng.random() * t_overlap
-
             city.tracker.track_social_mixing(self, h, location, distance, t_near)
             contact_condition = distance <= INFECTION_RADIUS and t_near > INFECTION_DURATION
+            infectee = None
+
+            # Conditions met for possible infection
             if contact_condition:
                 proximity_factor = 1
                 if INFECTION_DISTANCE_FACTOR or INFECTION_DURATION_FACTOR:
                     proximity_factor = INFECTION_DISTANCE_FACTOR * (1 - distance/INFECTION_RADIUS) + INFECTION_DURATION_FACTOR * min((t_near - INFECTION_DURATION)/INFECTION_DURATION, 1)
-                mask_efficacy = self.mask_efficacy * h.mask_efficacy
-
-                infectee = None
-                if self.is_infectious:
-                    ratio = self.asymptomatic_infection_ratio  if self.is_asymptomatic else 1.0
-                    p_infection = self.infectiousness * ratio * (1-mask_efficacy) * proximity_factor
-                    x_human = self.rng.random() < p_infection * CONTAGION_KNOB
-
-                    if x_human and h.is_susceptible:
-                        h.infection_timestamp = self.env.timestamp
-                        self.n_infectious_contacts+=1
-                        Event.log_exposed(h, self, self.env.timestamp)
-                        city.tracker.track_infection('human', from_human=self, to_human=h, location=location, timestamp=self.env.timestamp)
-                        # print(f"{self.name} infected {h.name} at {location}")
-                        infectee = h.name
-
-                elif h.is_infectious:
+                mask_efficacy = h.mask_efficacy # anything that got through their mask will likely get through ours
+                # # I am infectious 
+                # if self.is_infectious:
+                #     ratio = self.asymptomatic_infection_ratio  if self.is_asymptomatic else 1.0
+                #     p_infection = self.infectiousness * ratio * (1-mask_efficacy) * proximity_factor
+                #     x_human = self.rng.random() < p_infection * CONTAGION_KNOB
+
+                #     # Infection
+                #     if x_human and h.is_susceptible:
+                #         h.infection_timestamp = self.env.timestamp
+                #         self.n_infectious_contacts+=1
+                #         Event.log_exposed(h, self, self.env.timestamp)
+                #         city.tracker.track_infection('human', from_human=self, to_human=h, location=location, timestamp=self.env.timestamp)
+                #         # print(f"{self.name} infected {h.name} at {location}")
+                #         infectee = h.name
+
+                # Encountered human is infectious
+                if h.is_infectious:
```

Comment:
I think this will miss a case when `self` is infectious. This will not let self infect other humans already at the location.

Reply:
I don't think it will miss that; it will be taken care of by the loops of other humans. Call the current self Zach, who is infectious. He goes through a loop with Yiu and Xena. Then we do loops where Yiu and Xena each become "self", and they will see Zach as "h", who will infect them.

As long as we don't remove humans from locations when they're done looping.

Reply:
`[A] --> [A, B*] --> [A, B*, C]` by the above logic, B will infect C, but not A.

Reply:
[We had a long discussion on slack on this issue](https://covid19-task-force.slack.com/archives/C011C7KGPJB/p1586460189059800?thread_ts=1586458139.045000&cid=C011C7KGPJB).